codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -862,17 +862,15 @@ let injectedTaskReducer = (
           false
         }
         let timeRef = Hrtime.makeTimer()
-        await Db.sql->Postgres.beginSql(sql => {
-          Entities.allEntities->Belt.Array.map(entityMod => {
-            let module(Entity) = entityMod
-
-            sql->DbFunctions.EntityHistory.pruneStaleEntityHistory(
-              ~entityName=Entity.name,
-              ~safeChainIdAndBlockNumberArray,
-              ~shouldDeepClean,
-            )
-          })
-        })
+        let _ = await Promise.all(Entities.allEntities->Belt.Array.map(entityMod => {
+          let module(Entity) = entityMod
+          Db.sql->DbFunctions.EntityHistory.pruneStaleEntityHistory(
+            ~entityName=Entity.name,
+            ~safeChainIdAndBlockNumberArray,
+            ~shouldDeepClean,
+          )
+        }))
```

Comment:
Removed a transaction here:
1. It's not needed here
2. Possible to use multiple connections
3. So we have only one place where we use a transaction to make it easier to get rid of :)